collectfast/strategies/filesystem.py

```diff
@@ -11,3 +15,27 @@ def get_remote_file_hash(self, prefixed_path: str) -> Optional[str]:
             return self.get_local_file_hash(prefixed_path, self.remote_storage)
         except FileNotFoundError:
             return None
+
+
+class CachingFileSystemStrategy(
+        CachingHashStrategy[FileSystemStorage],
+        FileSystemStrategy,
+):
+    @lru_cache()
+    def get_local_file_hash(self, *args, **kwargs):
+        '''
+        caches the local file hash in memory so the hash is only computed once
+        per run and `post_copy_hook` method the does not duplicate work
+        '''
+        return super().get_local_file_hash(*args, **kwargs)
+    
+    def post_copy_hook(
+            self, path: str, prefixed_path: str, local_storage: Storage
+    ) -> None:
+        '''
+        sets the cached hash from the local file that was just copied to avoid
+        reading the remote file which may be stored on a networked file system
+        '''
+        key = self.get_cache_key(path)
+        value = self.get_local_file_hash(path, local_storage)
+        cache.set(key, value)
```

Comment:
I think it could make sense to implement this in `CachingHashStrategy` using a class var like `prime_cache_using_local_hash: ClassVar = False` and setting that to `True` in this class. What do you think?

Reply:
I was planning to refactor once I finish up my few issues.  I have an idea of what is required but I haven't seen it all come together yet.  There are a few other places that can simplify as well.

Reply:
Do you see any reason this behavior should be disabled?  It seems the best course of action in all cases as it is the purpose of `CachingHashStrategy`

Reply:
Yes, this definitely need to be disabled by default. This wont necessarily be the best course of action for all backends. For large files looking up the hash from s3 on the next run might be preferable to computing it locally.

Reply:
(sent from cell phone so forgive formatting)

Since should_copy_file() calls get_local_file_hash() to hash the local file on each run the local hash is memorized in the lru_cache on the method when this is called so there is no processing.

This improves performance on the s3 strategy by preventing a redundant http request (because the hash is already known when this is call) to s3 on subsequent runs.

Reply:
You're right.